### PR TITLE
[IMP] sales: add _create_invoices hook in sale.advance.payment.inv

### PR DESCRIPTION
**Pull request by IT-Ideas**
Prior to this commit it was not possible to get a hook from
sale.advance.payment.inv in order to get the created invoices.

The create_invoices def returns either an action or an ir.actions.act_window_close

The introduction of the _create_invoices def will allow to get the created invoices
from the wizzard.

task-2608812

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by xavierbol — CHANGES_REQUESTED** [review]
Thanks for your work, I have made a suggestion. ^^

File: addons/sale_timesheet/wizard/sale_make_invoice_advance.py [hunk 1]
```diff
@@ -24,26 +24,18 @@ def _default_invoicing_timesheet_enabled(self):
...
                 sale_orders.mapped('order_line')._recompute_qty_to_invoice(self.date_start_invoice_timesheet, self.date_end_invoice_timesheet)
 
-            sale_orders._create_invoices(final=self.deduct_down_payments, start_date=self.date_start_invoice_timesheet, end_date=self.date_end_invoice_timesheet)
-
-            if self._context.get('open_invoices', False):
-                return sale_orders.action_view_invoice()
-            return {'type': 'ir.actions.act_window_close'}
+            return sale_orders._create_invoices(final=self.deduct_down_payments, start_date=self.date_start_invoice_timesheet, end_date=self.date_end_invoice_timesheet)
 
         return super(SaleAdvancePaymentInv, self).create_invoices()
```
[the remark this answers is not in the document]
Comment (IT-Ideas) [reply addons/sale_timesheet/wizard/sale_make_invoice_advance.py]: Oupsie! Thanks for the review!

**Review by Feyensv — COMMENTED** [review]
LGTM :+1:
I noticed there is one flow you don't notice with this hook, when we auto-invoice sale orders after they are paid (sale/models/payment_transaction.py:_invoice_sale_orders()), but I cannot say if it's really a problem.

**Review by Feyensv — APPROVED** [review]

File: addons/sale_timesheet/wizard/sale_make_invoice_advance.py [hunk 2]
```diff
@@ -24,26 +24,18 @@ def _default_invoicing_timesheet_enabled(self):
...
 
-            sale_orders._create_invoices(final=self.deduct_down_payments, start_date=self.date_start_invoice_timesheet, end_date=self.date_end_invoice_timesheet)
-
-            if self._context.get('open_invoices', False):
-                return sale_orders.action_view_invoice()
-            return {'type': 'ir.actions.act_window_close'}
+            return sale_orders._create_invoices(final=self.deduct_down_payments, start_date=self.date_start_invoice_timesheet, end_date=self.date_end_invoice_timesheet)
 
-        return super(SaleAdvancePaymentInv, self).create_invoices()
+        return super()._create_invoices()
```
[the remark this answers is not in the document]
Comment (IT-Ideas) [reply addons/sale_timesheet/wizard/sale_make_invoice_advance.py:41]: Arf :/ Thanks!

**Review by xavierbol — APPROVED** [review]
LGTM! 🙂

**Comment by IT-Ideas** [comment]
@robodoo rebase-ff

**Review by xavierbol — APPROVED** [review]
LGTM, thanks 🙂

**Review by xavierbol — COMMENTED** [review]
Thanks for your work I have made a small comment. 🙂

File: addons/hr_timesheet/models/project.py [hunk 3]
```diff
@@ -183,6 +183,14 @@ def write(self, values):
                     project._create_analytic_account()
         return super(Project, self).write(values)
 
+    def name_get(self):
+        name_mapping = dict(super().name_get())
+        if len(self.env.context.get('allowed_company_ids', [])) > 1:
+            for project in self:
+                if project.is_internal_project:
+                    name_mapping[project.id] = f'{name_mapping[project.id]} - {project.company_id.name}'
+        return list(name_mapping.items())
```
[the remark this answers is not in the document]
Comment (IT-Ideas) [reply addons/hr_timesheet/models/project.py:192]: Yep I agree!

**Comment by IT-Ideas** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**